src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -618,21 +618,45 @@ await foreach (ConfigurationSetting setting in client.GetConfigurationSettingsAs
                 }).ConfigureAwait(false);
             }
 
-                foreach (var loadOption in _options.KeyValueSelectors)
+            foreach (KeyValueSelector loadOption in _options.KeyValueSelectors)
+            {
+                IAsyncEnumerable<ConfigurationSetting> settingsEnumerable;
+
+                if (string.IsNullOrEmpty(loadOption.SnapshotName))
+                {
+                    settingsEnumerable = client.GetConfigurationSettingsAsync(
+                        new SettingSelector
+                        {
+                            KeyFilter = loadOption.KeyFilter,
+                            LabelFilter = loadOption.LabelFilter
+                        },
+                        cancellationToken);
+                }
+                else
                 {
-                    var selector = new SettingSelector
+                    ConfigurationSettingsSnapshot snapshot = await client.GetSnapshotAsync(loadOption.SnapshotName);
+
+                    if (snapshot != null)
                     {
-                        KeyFilter = loadOption.KeyFilter,
-                        LabelFilter = loadOption.LabelFilter
-                    };
+                        if (snapshot.CompositionType == null || !snapshot.CompositionType.ToString().Equals("key"))
+                        {
+                            throw new ArgumentException($"{nameof(snapshot.CompositionType)} for the selected snapshot must be 'key', found '{snapshot.CompositionType}'.");

```

Comment:
It's not an argument exception. No passed argument is invalid. It's more of an invalid operation exception.

Reply:
Hi, I just tried the latest preview version 7.0.0-preview.2 and tried to load a snapshot that has the key+label composition type and got this exception. Is the key+label just not implemented yet? I assume it will be since it's a little strange to be able to create those snapshots but not load them or am I missing something?